### DISCUSSION TITLE
What is the best way to get the files path of the NextJS's public folder files from a nodejs express server?

### DISCUSSION THREAD

**Discussion by fgarrec0397** [vercel/next.js#47545] [Help]
### Summary

 _The NextJS public folder is used to simplify the issue, but in fact the public folder architecture is not cast in stone_

I'm implementing a file explorer in my monorepo with Nextjs and I need to find all the files in the public folder to display it on the front-end.

I found a good approach in [this article](https://medium.com/@boris.poehland.business/next-js-api-routes-how-to-read-files-from-directory-compatible-with-vercel-5fb5837694b9) to achieve what I needed and it is working well.

However, in the article it uses the back-end api routing of Nextjs which I don't want to use since I already have my express server running on another port and I want to regroup all my routing under this server for obvious reasons. 

I refactored the article code to match my architecture and it worked.

My question is **Does the code below is a good approach to get the all the files in the NextJS's  public folder from an express server under a same monorepo?**

_This is a basic representation of my monorepo architecture_
```
root
    - apps
        - admin (nextjs app)
        - admin-server (express server app)
    - packages
        - all my packages...
```

_This is my getFiles controller_
```
import { Request, Response } from "express";
import path from "path";
import fs from "fs";

export const getFiles = async (request: Request, result: Response) => {
    const dirRelativeToPublicFolder = "assets";
    const dir = path.resolve("../admin", "public", dirRelativeToPublicFolder); // 

    const filenames = fs.readdirSync(dir);

    const images = filenames.map((name) => path.join("/", dirRelativeToPublicFolder, name));

    result.statusCode = 200;
    result.json(images);
};

```

Please ask for any more information or to clarify anything in my question.

**Comment by icyJoseph** [comment]
- https://nextjs.org/docs/api-reference/data-fetching/get-static-props#reading-files-use-processcwd

I think you should use `process.cwd()` though, it is a Node.js built-in, which tells you where the `node <something>` command was called, and relative to that you can find whatever file/folder you wish.

**Reply by fgarrec0397** [reply]
Thanks for the response!

It seems to be a good solution for when you use the backend of Nextjs, but in my case I use an Express server on the side that read the public folder and I was wondering if it was a good approach as mentionned in the question.
